src/walk.rs

```diff
@@ -226,63 +231,98 @@ fn spawn_receiver(
                 .unwrap_or_else(|| time::Duration::from_millis(100));
 
             let stdout = io::stdout();
-            let mut stdout = stdout.lock();
+            let stdout = stdout.lock();
+            let mut stdout = io::BufWriter::new(stdout);
 
             let mut num_results = 0;
 
-            for worker_result in rx {
-                match worker_result {
-                    WorkerResult::Entry(value) => {
-                        match mode {
-                            ReceiverMode::Buffering => {
-                                buffer.push(value);
-
-                                // Have we reached the maximum buffer size or maximum buffering time?
-                                if buffer.len() > MAX_BUFFER_LENGTH
-                                    || time::Instant::now() - start > max_buffer_time
-                                {
-                                    // Flush the buffer
-                                    for v in &buffer {
-                                        output::print_entry(
-                                            &mut stdout,
-                                            v,
-                                            &config,
-                                            &wants_to_quit,
-                                        );
+            if config.interactive_terminal {
+                let mut buffer = Vec::with_capacity(MAX_BUFFER_LENGTH);
+                for worker_result in rx {
+                    match worker_result {
+                        WorkerResult::Entry(path) => {
+                            match mode {
+                                ReceiverMode::Buffering => {
+                                    buffer.push(path);
+
+                                    // Have we reached the maximum buffer size or maximum buffering time?
+                                    if buffer.len() > MAX_BUFFER_LENGTH
+                                        || time::Instant::now() - start > max_buffer_time
+                                    {
+                                        // Flush the buffer
+                                        for path in &buffer {
+                                            output::print_entry(
+                                                &mut stdout,
+                                                path,
+                                                &config,
+                                                &wants_to_quit,
+                                            );
+                                        }
+                                        buffer.clear();
+                                        let r = stdout.flush();
```

Comment:
Perhaps this chunk of code:

```rust
                                       let r = stdout.flush();
                                        if r.is_err() {
                                            // Probably a broken pipe. Exit gracefully.
                                            process::exit(ExitCode::GeneralError.into());
                                        }
```

should be factored out into a function, since it is repeated a few times?

Reply:
Taking my other comment into account, it could potentially be something like:

```
fn maybe_flush<W: Write>(config: &Arc<Options>, stdout: &mut W>) {
   if config.interactive_terminal {
     if stdout.flush().is_err() {
       process::exit(ExitCode::GeneralError.into());
    }
  }
}
```